templates/Elegant/compare.tmpl

```diff
@@ -33,6 +33,9 @@
     </table>
     [websvn:compare_endform]
     </div>
+    [websvn-test:warning]
+      <div id="warning">[websvn:warning]</div>
+    [websvn-else]
```

Comment:
I suggest removing that ELSE-stuff from ALL templates: If there's a warning, it's of course fine to print that. But that shouldn't prevent any other valid output. Remember that the templates you just changed DIDN'T print the warning, but only the OTHER content and nobody cared, because things seemed to have worked. :-) By introducing the ELSE, this gets changed to NEVER print the content even if the warning isn't too important at all.

Reply:
@ams-tschoening So you are proposing that the warning should be printed additionally to the regular output?

Reply:
With this particular change i think warning will only be generated in case the revision number is out of bounds. 
Meaning we try to compare some location or file where both revision entered are out of bounds. 

In all other cases there will be either a difference that will be shown or it will be blank.
I have captured some scenarios with this implementation below(I am putting images because it is faster):
Both `$rev1` and `$rev2` out of bounds
![image](https://user-images.githubusercontent.com/4973182/90340496-3fa67f80-e016-11ea-8944-df21be616262.png)

One of the `$rev1` or `$rev2` out of bounds
![image](https://user-images.githubusercontent.com/4973182/90340539-8d22ec80-e016-11ea-940d-c65c72666605.png)

Both `$rev1` and `$rev2` are under bounds but no changes between revision
![image](https://user-images.githubusercontent.com/4973182/90340577-cb201080-e016-11ea-8801-3f2a249eb048.png)

Both `$rev1` and `$rev2` are under bounds with changes between revision
![image](https://user-images.githubusercontent.com/4973182/90340589-ea1ea280-e016-11ea-9d90-a48ce86076f8.png)

Reply:
Will Subversion tell you that the rev is out of bounds?

Reply:
What we will get in the log is a null value. If we dump the history variable we get a null value at log entry node. That is how we come to know it is out of bounds.
`object(SVNLog)#5 (3) { ["entries"]=> array(0) { } ["curEntry"]=> NULL ["path"]=> string(14) "/trunk/main.py" }`
In all other scenarios we always get a "currEntry" node.

Reply:
So this is OK for my. Shit it, nothing out.

Reply:
So what's your argument in the end? That printing the warning additionally is NOT necessary in your test cases? Might be, but what's the problem with printing it ADDITIONALLY? Remember that that's exactly how the current implementation works: It stores some warning and proceeds to generate further content. If that content' doesn't make much sense, bad luck, but because of the warning one knows. If that content DOES make sense, good for the user, but the warning makes one have a more detailed look as well. Remember it's a warning only, almost ALL software I know of continues work after logging/showing/... warnings.

Besides that, if one revision is out of bands and you are able to detect that, this qualifies for a warning in my opinion as well. Not only if both revisions are out of band.

Reply:
Yeah.. My argument was to convert the warning to error. But now reading your argument maybe that is not the best way forward. It really makes no difference to user if he sees warning in addition to content.  
This makes a way better sense than the direction i was thinking in.
Okay i have pushed a change in template where i removed the else condition and added a success check in all the templates.